feat(steamvr): auto-unblock ALVR driver on startup

skryvel opened the pull request:
I'm really not sure what the path is on Windows. I found some references but not a full path, and especially it varies.

This will go into `steamvr.vrsettings` and toggle `blocked_by_safe_mode` if needed for the ALVR driver. Should fail gracefully as long as the file exists.

Windows is not implemented because:
 - on Discord the point was raised that Windows SteamVR gets/causes much less trouble with this;
 - the path gets more complicated and no source of truth on the web was found that felt safe to quickly implement.

zmerp commented on alvr/server_io/src/openvrpaths.rs:
```diff
@@ -26,6 +26,24 @@ fn openvr_source_file_path() -> Result<PathBuf> {
 
+pub fn openvr_settings_file_path() -> Result<PathBuf> {
+    let path = if cfg!(windows) {
+        // N.B. if ever implementing this: given Steam can be installed on another
+        // drive, etc., this should probably start by looking at Windows registry keys.
+        unimplemented!() // Original motive for implementation had little reason for Windows.
```
Isn't it possible to obtain this from the `openvrpaths.vrpath` file? Another option is to use the default windows path (in the C drive) and gracefully fail if path is not found, by not unblocking the alvr driver

skryvel replied:
> vrpath
Potentially it always matches the `config` entry. I don't know how to be sure but I can do that if told to.
> crash
There's another conditional in the code actually using it, but I agree.

skryvel replied:
I changed the unimplemented to a bail, since this already is `Result`.

zmerp reviewed (changes requested):
To avoid crashes, all `unwrap()`s should be removed and replaced with proper error handling

skryvel commented:
> To avoid crashes, all `unwrap()`s should be removed and replaced with proper error handling

Fixed. Feel free to point me to any other samples or let me know of any nits -- I'm relatively new to Rust in such a code-base.

zmerp reviewed (approved):
Overall this looks good to me. Someone else on linux like @Meister1593  should also review and test this PR before merging.

Meister1593 commented on alvr/dashboard/src/steamvr_launcher/mod.rs:
```diff
@@ -108,6 +161,10 @@ impl Launcher {
         }
 
+        if let Err(err) = unblock_alvr_driver() {
+            debug!("Failed to unblock ALVR driver: {:?}", err);
```
This probably should be a `warn!` or `error!`

Meister1593 commented on alvr/server_io/src/openvrpaths.rs:
```diff
@@ -26,6 +26,24 @@ fn openvr_source_file_path() -> Result<PathBuf> {
     }
 }
 
+pub fn openvr_settings_file_path() -> Result<PathBuf> {
```
rename it to `steamvr_settings_file_path` (since, well, it is steamvr settings and not openvr's config file)

Meister1593 reviewed (changes requested):
Simulated scenario (put crash state into same file), works fine, though i rarely get any real crash events nowadays.
Only small changes needed, otherwise looks good

The-personified-devil reviewed (approved):
Haven't tested it (and I honestly can't be bothered to force steamvr to crash), but lgtm (tho quite verbose, which might honestly be good considering this is messing with steamvr's rather fragile internals).

gabriele2000 commented:
> Simulated scenario (put crash state into same file), works fine, though i rarely get any real crash events nowadays. Only small changes needed, otherwise looks good

Rarely?
SteamVR is as fragile as touching 20 year-old code, it crashes a lot, it's riddled with issues... consider yourself lucky
